### PR TITLE
Issue #23 (Update RSXA)

### PR DESCRIPTION
**Pull request by nsmohan**
### Changes to fix Issue: #23 

- #### RSXA.c:
          - Create new Function RSXA_init(). This function is to be called once and it will read the 
             RSXA.json file and parse it. 
           - RSXA_get_mode() no longer reads the RSXA.json file it simply parses the json structure and 
             fills the sim_mode flag
           - Updated comments for Doxygen Format

- #### MTDR.c & PCA9685.c:
              - Removed all calls to RSXA_get_mode() and made it so the sim_mode is passed in as an 
                argument each time
              - Update MTDR_init() so we move the motor to the home position upon initialization. 
              - Updated comments for Doxygen Format

- #### CAM_MOTOR_CTRL.cpp:
            - Updated the constructor so it now makes the call to initialize RSXA.json and store contents 
               in a structure in the constructor. 
             - Pass sim_mode to all MTDR functions
             - Updated comments for Doxygen Format

- #### camctrl.cpp:
           - Updated comments for Doxygen Format

- #### Additional Cleanup:
           - Updated Tests so they pass given the changes to the interfaces
           - Move python libraries to a new folder (lib_py)
           - Added a new Makefile in tst directory which creates a symbolic link in the tst directory with Obj and lib_py

### Note ** This may also fix Issue: #19. Issue #19 will go into monitoring after this fix has been merged in.

### CODE REVIEW

File: lib/CAM_MOTOR_CTRL.cpp [hunk 1]
```diff
@@ -13,57 +17,70 @@ __copyright__   = "Copy Right 2019. NM Technologies" */
...
+     */
+
+    /* Initialize Varibles */
     this->horizontal_motor = CAM_HRZN_MTR;
     this->vertical_motor = CAM_VERT_MTR;
+    this->settings = {0};
+    this->hw_settings_obj = {0};
+
+    /* Get Hardware Settings */
+    RSXA_init(&this->hw_settings_obj);
```
Comment (nsmohan) [inline lib/CAM_MOTOR_CTRL.cpp]: Change to: (&(this->hw ..)

File: lib/RSXA.c [hunk 2]
```diff
@@ -18,79 +23,103 @@ __copyright__   = "Copy Right 2019. NM Technologies" */
...
-    RSXA_hw hw          = {0};
     char *file_content;
 
     /* Read and parse the Json settings file */
     result = NMT_stdlib_read_file(RS_SETTINGS_PATH, &file_content);
 
     if (result == OK)
-        result = RSXA_parse_json(file_content, &hw);
+        result = RSXA_parse_json(file_content, hw);
+
```
Comment (nsmohan) [inline lib/RSXA.c]: Remove extra space

File: Makefile [hunk 3]
```diff
@@ -7,14 +7,19 @@ export BIN_DIR     = $(PROJ_DIR)/bin
 export OBJ_DIR     = $(PROJ_DIR)/Obj
 export INC_DIR     = $(PROJ_DIR)/inc
 export LIB_DIR     = $(PROJ_DIR)/lib
+export LIBPY_DIR     = $(PROJ_DIR)/lib_py
```
Comment (ChemistryMickey) [inline Makefile]: Spacing.
Aside from that, having a LIBPY directory sounds like a good idea if you're doing the tkinter now.

Comment (nsmohan) [reply Makefile]: Fixed.

**Comment by nsmohan** [comment]
I am not sure if this will resolve the segfaults but it will definitely help it. 

Also -> remove all instances of nmohan in the tst directory.

**Pull request merged**